src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -180,8 +205,30 @@ private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr arg)
 
                         using (X509Certificate2 leafCert = new X509Certificate2(leafCertPtr))
                         {
-                            success = chain.Build(leafCert);
-                            AddChannelBindingToken(leafCert, arg);
+                            var serverCallback = easy._handler._serverCertificateValidationCallback;
+                            if (serverCallback == null)
+                            {
+                                SslPolicyErrors errors = CertificateValidation.BuildChainAndVerifyProperties(
+                                    chain, leafCert, checkCertName: false, hostName: null);
+                                success = errors == SslPolicyErrors.None;
```

Comment:
@stephentoub If that's something that libcurl isn't doing automatically we'll want to backfill that to rc2.  But since that functionality isn't (or at least wasn't) part of OpenSSL I'd expect it to be in curl already.